### PR TITLE
ExerciseActivity and Amazon SNS

### PR DESCRIPTION
**Pull request by Dantemss**
Copies the grade from the GradingEvent, but keeps track of the AnswerEvents indirectly.

### CODE REVIEW

File: app/models/task.rb [hunk 1]
```diff
@@ -20,4 +20,20 @@ class Task < ActiveRecord::Base
...
+    # Try to find duplicates in the DB
+    duplicate_task = Task.find_by(identifier: identifier,
+                                  resource: resource,
+                                  trial: trial)
+
+    # Return pre-existing task if found
+    return duplicate_task unless duplicate_task.nil?
+
+    # No duplicates. Attempt to save and return self.
+    save
```
Comment (kevinburleigh75) [inline app/models/task.rb]: Do we need to handle the case where `save` fails?

File: app/representers/api/v1/grading_event_representer.rb [hunk 2]
```diff
@@ -5,8 +5,17 @@ class GradingEventRepresenter < EventRepresenter
...
+
+               # If an identifier is submitted as a grader,
+               # convert it to a research_label for researchers
+               Doorkeeper::AccessToken.find_by(token: grader)
+                                      .try(:resource_owner)
+                                      .try(:research_label) || grader
+             },
              schema_info: {
-               description: 'Who graded this Task'
+               description: 'The person grading this Task'
```
Comment (kevinburleigh75) [inline app/representers/api/v1/grading_event_representer.rb]: Are all graders people?  Does this need to be re-worded?

Comment (Dantemss) [inline app/representers/api/v1/grading_event_representer.rb]: True. I allow graders to be any string. I should change this description.

File: app/models/task.rb (hunk 1 above)
Comment (Dantemss) [inline app/models/task.rb]: This is currently handled by the `transfer_errors_from event.task` in the CreateEvent routine.

But if there are better ways to handle this, I'm open to suggestions.

**Pull request merged**